src/server/status/index.js

```diff
@@ -18,7 +19,11 @@ export default function (kbnServer, server, config) {
   server.route(wrapAuth({
     method: 'GET',
     path: '/api/status',
-    handler: function (request, reply) {
+    handler: async function (request, reply) {
+      const stats = getStats(
+        server.config().get('kibana.index'),
+        request,
+        server.plugins.elasticsearch.getCluster('admin').callWithRequest);
```

Comment:
My concern with having `callWithRequest` used is that the calling user credentials might not have privilege to read the `.kibana` index. We may be dealing with a Monitoring-only user, for example. We talked about being safe in letting the API return the right data for any user, regardless of role, because there's no confidential information in this data - it's just counts of things. 

I'm not sure I follow the point about "we could make the class interface for fetching optionally use an admin client" - does that go along with what Tyler suggests above? We just have to be sure that the API that Monitoring uses doesn't require any special roles for the calling user.